include/pybind11/chrono.h

```diff
@@ -161,9 +161,17 @@ template <typename Duration> class type_caster<std::chrono::time_point<std::chro
         // > If std::time_t has lower precision, it is implementation-defined whether the value is rounded or truncated.
         // (https://en.cppreference.com/w/cpp/chrono/system_clock/to_time_t)
         std::time_t tt = system_clock::to_time_t(time_point_cast<system_clock::duration>(src - us));
+
+        // std::localtime returns a pointer to a static internal std::tm object on success,
+        // or null pointer otherwise
+        // (https://en.cppreference.com/w/cpp/chrono/c/localtime)
+        std::tm *localtime_ptr = std::localtime(&tt);
+        if (!localtime_ptr)
+            pybind11_fail("pybind11::chrono: Unable to represent system_clock in local time");
```

Comment:
`pybind11_fail` is meant for internal errors, so this is not correct. Maybe throw `py::cast_error`? Maybe have a look at what other casters do, please.

Reply:
How about now?

Reply:
If the CI wants to turn green, this seems better, indeed. Thanks! :-)

Reply:
I am not sure what broke in `test_recarray` in test_numpy_dtypes.py on CentOS 7. I will try debugging it, now that it has failed twice.

Reply:
Very weird that it would be related, though

Reply:
It's not, it's started breaking everywhere.